### PR TITLE
EG-1168, Outlaw page redirects except after input, enable return to page id on input error (sync from viewer)

### PR DESCRIPTION
**Pull request by cmrhmrc**

### CODE REVIEW

File: test/services/ApprovalServiceSpec.scala [hunk 1]
```diff
@@ -192,7 +192,7 @@ class ApprovalServiceSpec extends BaseSpec with MockFactory {
     "the processCode exists for another process in the publishedCollection" should {
       "return a DuplicateKeyException" in new Test {
 
-        val expected: RequestOutcome[String] = Right(validId)
+        //val expected: RequestOutcome[String] = Right(validId)
```
Comment (LesWilson) [inline test/services/ApprovalServiceSpec.scala:195]: Can we remove this line if no longer required

File: test/services/PageBuilderSpec.scala [hunk 2]
```diff
@@ -122,6 +122,34 @@ class PageBuilderSpec extends BaseSpec with ProcessJson with StanzaHelper {
...
 
+    "detect SharedDataInputStanza error" in {
+      val flow = Map(
+        Process.StartStanzaId -> PageStanza("/url", Seq("1"), true),
+        "1" -> InstructionStanza(0, Seq("2"), None, false),
+        "2" -> InputStanza(Currency, Seq("4"), 0, Some(0), "Label", None, false),
+        "4" -> Choice(ChoiceStanza(Seq("5","end"), Seq(ChoiceStanzaTest("[label:label]", LessThanOrEquals, "8")), false)),
+        "5" -> PageStanza("/url2", Seq("1"), true),
+        "6" -> InstructionStanza(0, Seq("2"), None, false),
+        "2" -> InputStanza(Currency, Seq("4"), 0, Some(0), "Label", None, false),
```
Comment (LesWilson) [inline test/services/PageBuilderSpec.scala:133]: Can this flow be generated in Ocelot?  Duplicate id of "2"

Comment (cmrhmrc) [reply test/services/PageBuilderSpec.scala:133]: See viewer comment

File: test/services/ApprovalServiceSpec.scala (hunk 1 above)
Comment (cmrhmrc) [reply test/services/ApprovalServiceSpec.scala:195]: removed

**Review by LesWilson — APPROVED** [review]

**Pull request merged**